state-reconstruct-fetcher/src/lib.rs

```diff
@@ -1,9 +1,660 @@
 #![feature(array_chunks)]
 #![feature(iter_next_chunk)]
+
 pub mod api_support;
 pub mod blob_http_client;
 pub mod constants;
-pub mod database;
-pub mod l1_fetcher;
 pub mod metrics;
 pub mod types;
+
```

Comment:
Why move everything into lib.rs? Shouldn't L1Fetcher be defined in l1_fetcher.rs?

Reply:
I think it makes sense since the previous `lib.rs` only contained re-exports of those modules and `L1Fetcher` should be the main export (it's currently tied up a bit with the dependancy on `mod types`) however. In the end, I don't feel particularly strongly about this though.

Reply:
well, I think re-exports is exactly what lib.rs should contain - mixing in implementation just lowers its cohesion...